test/integration_test/basic_test.go

```diff
@@ -107,6 +107,34 @@ var testStorageClusterBasicCases = []types.TestCase{
 		},
 		TestFunc: BasicUpgradeStorageCluster,
 	},
+	{
+		TestName:        "BasicUpgradeStorageClusterWithAllComponents",
+		TestrailCaseIDs: []string{"C50241"},
+		TestSpec: func(t *testing.T) interface{} {
+			objects, err := ci_utils.ParseSpecs("storagecluster/storagecluster-with-all-components.yaml")
+			require.NoError(t, err)
+			cluster, ok := objects[0].(*corev1.StorageCluster)
+			require.True(t, ok)
+			cluster.Name = "upgrade-stc-with-all-components-test"
+			alertManagerSecret, err := ci_utils.ParseSpecs("monitoring/alertmanager-secret.yaml")
```

Comment:
can we parse and return the stc object only in this TestSpec function here without doing the alert manager secret creation inside? we can do this in the test function after getting the stc, check stc spec if alertmanager is enabled, then parse it then create.

Reply:
Moved